pkg/volume/configmap/configmap.go

```diff
@@ -217,7 +217,16 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		len(configMap.Data),
 		totalBytes)
 
-	payload, err := MakePayload(b.source.Items, configMap, b.source.DefaultMode, optional)
+	defaultMode := b.source.DefaultMode
+	if defaultMode == nil {
+		mode := int32(0400)
```

Comment:
why the defaultMode is int32(0400) here?
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L1625
`// value between 0 and 0777. Defaults to 0644.`

Reply:
those defaults are applied by the API server, and mean that defaultMode will likely never be nil. in the case where defaultMode *is* nil, this code is using the same minimal permissions as SetVolumeOwnership

Reply:
on further thought, we can depend on these not being nil... removed this secondary defaulting